src/fsharp/lex.fsl

```diff
@@ -405,9 +405,13 @@ rule token args skip = parse
        UINT64(n) } 
 
  | nativeint  
-     { try 
-           NATIVEINT(int64 (removeUnderscores (lexemeTrimRight lexbuf 1)))
-       with _ ->  fail args lexbuf (FSComp.SR.lexOutsideNativeSigned()) (NATIVEINT(0L)) } 
+     { let s = removeUnderscores (lexemeTrimRight lexbuf 1)
+       // Allow <max_nativeint+1> to parse as min_nativeint.  Stupid but allowed because we parse '-' as an operator. 
+       if s = "9223372036854775808" then NATIVEINT(-9223372036854775808L, true) else
```

Comment:
Magic values are less than ideal:

Perhaps precompute the values as static values somewhere maybe :
let minNative = System.Int64.MinValue
let minNativeWithoutSign =
    let s= minNative.ToString(System.Globalization.CultureInfo.InvariantCulture)
    s.Substring(1, s.Length - 1)

Reply:
Are these .NET calls available at this stage? looking at the surrounding code it also has magical values which would lead me to believe some things aren't available

Reply:
@cartermp , yes they are available.  They are values in mscorlib et al.  Indeed, magical values are everywhere in the compiler, that doesn't make them anything other than less than ideal.

Reply:
@KevinRansom I can certainly update, I actually considered it, but chose the "so as neighboring code" approach. Ik not a fan of the magic values either, glad you brought it up :). 

I'll check globally if this can be fixed easily by using the predefined constants.

Reply:
Is there a place for such globally used constants I should add this too?

Reply:
Just somewhere near the top of the file, they aren't really generally useful.

Reply:
Done, not just for this one, but for all integers. @kevinransom, this is ready for rereview :).